src/crate/client/sqlalchemy/compiler.py

```diff
@@ -244,6 +244,41 @@ def visit_any(self, element, **kw):
             self.process(element.right, **kw)
         )
 
+    def visit_ilike_case_insensitive_operand(self, element, **kw):
+        """
+        Use native `ILIKE` operator, like PostgreSQL's `PGCompiler`.
+        """
+        return element.element._compiler_dispatch(self, **kw)
+
+    def visit_ilike_op_binary(self, binary, operator, **kw):
+        """
+        Use native `ILIKE` operator, like PostgreSQL's `PGCompiler`.
+        """
+        escape = binary.modifiers.get("escape", None)
+
+        return "%s ILIKE %s" % (
+            self.process(binary.left, **kw),
+            self.process(binary.right, **kw),
+        ) + (
+            " ESCAPE " + self.render_literal_value(escape, sqltypes.STRINGTYPE)
+            if escape
+            else ""
+        )
```

Comment:
Does CrateDB support `ESCAPE` in this context?

Reply:
No, 
```
UnsupportedFeatureException[ESCAPE is not supported.]
```
Maybe we should raise a feature request?

How were % and _ handled previously in the `lower() like lower()` implementation?

Reply:
I was hoping that you can explain. ;]

Reply:
Shall we just remove the corresponding code, in order to be able to bring in the patch, and create an issue at crate/crate to track concerns of the `ESCAPE` operator/keyword? I guess it can also be used on other statements/clauses?

Reply:
Yes please, it seems that postgres supports defining a custom escape char seq:
https://www.postgresql.org/docs/15/functions-matching.html
https://stackoverflow.com/questions/196626/escaping-in-a-ilike-query
which is not the case for CrateDB.